RAGchain/reranker/bm25.py

```diff
@@ -0,0 +1,49 @@
+from typing import List
+
+from rank_bm25 import BM25Okapi
+from tqdm import tqdm
+from transformers import AutoTokenizer
+
+from RAGchain.reranker.base import BaseReranker
+from RAGchain.schema import Passage
+
+
+class BM25Reranker(BaseReranker):
+    """
+    BM25Reranker is a reranker based on BM25.
+    You can rerank the passages with the instruction using BM25Reranker.
+    """
+
+    def __init__(self, tokenizer_name: str = "gpt2", *args, **kwargs):
+        self.tokenizer = AutoTokenizer.from_pretrained(tokenizer_name)
+
+    def rerank(self, query: str, passages: List[Passage]) -> List[Passage]:
+        contents: List[str] = [passage.content for passage in passages]
+
+        # tokenize content for bm25 instance
+        tokenized_content: List[str] = [self.__tokenize([content])[0] for content in tqdm(contents)]
```

Comment:
I think you can tokenize list of string with __tokenize() method without using for-loop comprehension.

Reply:
Is the reason for to to make the code easier to read?